src/azul/service/responseobjects/hca_response_v5.py

```diff
@@ -238,15 +242,71 @@ def validate(s: str) -> str:
             column_value = self.column_joiner.join(sorted(set(column_value)))
             row[column_name] = column_value
 
+    def _get_header(self, object_key):
+        try:
+            header = self.storage_service.head(object_key=object_key)
+        except self.storage_service.client.exceptions.ClientError as e:
+            if int(e.response['Error']['Code']) == 404:
+                return None
+            else:
+                raise e
+        else:
+            return header
+
+    @classmethod
+    def _get_seconds_until_expire(cls, header: dict) -> float:
+        """
+        Get the number of seconds before a cached manifest is past its expiration.
+
+        :param header: A storage service object header dict
+        :return: time to expiration in seconds
+        """
+        # example x-amz-expiration = 'expiry-date="Fri, 21 Dec 2012 00:00:00 GMT", rule-id="Rule for testfile.txt"'
+        date_time_now = datetime.now(timezone.utc)
+        x_amz_expiration = parse_dict_header(header['x-amz-expiration'])
+        expiry_date = parse_date(x_amz_expiration['expiry-date']).replace(tzinfo=date_time_now.tzinfo)
```

Comment:
Forcing the timezone is not a good idea since we don't know for sure that the TZ offset in header is UTC. This is actually compensating for a bug in `parse_date` (which somehow parses out the TZ offset but then returns a offset-naive datetime object). See my commit for a safer solution.